status/condition_set_test.go

```diff
@@ -105,4 +107,49 @@ var _ = Describe("Conditions", func() {
 		Expect(testObject.StatusConditions().IsTrue(ConditionTypeFoo, ConditionTypeBaz)).To(BeTrue())
 		Expect(testObject.StatusConditions().IsTrue(ConditionTypeFoo, ConditionTypeBar, ConditionTypeBaz)).To(BeTrue())
 	})
+	It("should validate status condition on unstructured object status is false", func() {
+		testObject := &unstructured.Unstructured{}
+		testObject.SetUnstructuredContent(createStatusCondition("False"))
+		conditions := status.FromUnstructured(testObject).StatusConditions()
+		Expect(conditions.IsTrue(status.ConditionReady)).To(BeFalse())
+	})
+	It("should validate status condition on unstructured object status is true", func() {
+		testObject := &unstructured.Unstructured{}
+		testObject.SetUnstructuredContent(createStatusCondition("True"))
+		conditions := status.FromUnstructured(testObject).StatusConditions()
+		Expect(conditions.IsTrue(status.ConditionReady)).To(BeTrue())
+	})
+	It("should set condition on unstructured object", func() {
+		testObject := &unstructured.Unstructured{}
+		testObject.SetUnstructuredContent(createStatusCondition("True"))
+		conditions := []status.Condition{
+			{
+				Type:    status.ConditionReady,
+				Status:  metav1.ConditionFalse,
+				Reason:  "reason",
+				Message: "message",
+			},
+		}
+		status.FromUnstructured(testObject).SetConditions(conditions)
+		c, found, err := unstructured.NestedSlice(testObject.Object, "status", "conditions")
+		Expect(err).To(BeNil())
+		Expect(found).To(BeTrue())
+		Expect(len(c)).To(BeEquivalentTo(2))
+	})
 })
+
+func createStatusCondition(status string) map[string]interface{} {
```

Comment:
```suggestion
func createUnstructuredStatusConditions(status string) map[string]interface{} {
```

Reply:
```suggestion
func createStatusCondition(status string) unstructured.Unstructured {
```

Should this return `unstructured.Unstructured` to more closely model the code behavior? I think it's just a couple more lines that you are adding in each test case anyways